cozmo_soar.py

```diff
@@ -648,6 +654,153 @@ def on_input_phase(self, input_link: sml.Identifier):
+            
+        # we only want to send information on blocks whose locations we know.  in navmap, it initializes blocks at position (0, 0, 0) even if it cannot see them.  this function checks to see if this is the "init_block" state
+        def block_init(val):
+            pos = val.pose.position
+            return val.is_visible is False and pos.x == 0.00 and pos.y == 0.00 and pos.z == 0.00
+
+        # create and populate lists to add, delete, and change elements in SVS
+        navmap = self.w.__dict__['_objects']
+        buff = self.svs_buffer
+        change_list = []
+        add_list = []
+        tag_list = [] # used for is_visible tags
+        
+        # deepcopy val from navmap
+        def deepcopy(val):
+            d = {}
+            n = SimpleNamespace(**d)
+            n.object_id = val.object_id
+            n.is_visible = val.is_visible
+            n.pose = val.pose
+            return n
+
+        # loop through navmap items checking for changes / additions and add to lists to be processed
+        for _, val in navmap.items():
+            oid = val.object_id
+            if not block_init(val):
+                if oid in buff:
+                    if not equal(val, buff[oid]):
+                        change_list.append(val)
+                        if val.is_visible is not buff[oid].is_visible:
+                            tag_list.append(f'tag change {oid} is_visible {val.is_visible}')
+                        buff[oid] = deepcopy(val)
+                else:
+                    if val.is_visible:
+                        add_list.append(val)
+                        tag_list.append(f'tag add {oid} is_visible True')
+                        buff[oid] = deepcopy(val)
+            
+        def get_obj_str(obj, str_type):
+            position = obj.pose.position
+            rotation = obj.pose.rotation
+            pos_arr = [position.x, position.y, position.z]
+            quat = (rotation.q0, rotation.q1, rotation.q2, rotation.q3)
+            oid = obj.object_id
+
+            def quaternion_to_euler(quat):
+                """
+                credit: https://stackoverflow.com/questions/53033620/how-to-convert-euler-angles-to-quaternions-and-get-the-same-euler-angles-back-fr?rq=1
+                """
+                import math
+                (w, x, y, z) = quat
+                t0 = +2.0 * (w * x + y * z)
+                t1 = +1.0 - 2.0 * (x * x + y * y)
+                X = math.atan2(t0, t1)
+
+                t2 = +2.0 * (w * y - z * x)
+                t2 = +1.0 if t2 > +1.0 else t2
+                t2 = -1.0 if t2 < -1.0 else t2
+                Y = math.asin(t2)
+
+                t3 = +2.0 * (w * z + x * y)
+                t4 = +1.0 - 2.0 * (y * y + z * z)
+                Z = math.atan2(t3, t4)
+                return (X, Y, Z)
+
+            eul = quaternion_to_euler(quat)
+
+            if type(obj).__name__ == '_proxy_LightCube':
```

Comment:
Use `isinstance` here, it's cleaner and more Pythonic

```suggestion
            if isinstance(obj, cozmo.objects.LightCube):
```